src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2919,6 +2914,23 @@ private BoundExpression BindDelegateCreationExpression(ObjectCreationExpressionS
                     return new BoundDelegateCreationExpression(node, boundLambda, methodOpt: null, isExtensionMethod: false, type: type, hasErrors: !conversion.IsImplicit);
                 }
 
+                else if (analyzedArguments.HasErrors)
```

Comment:
Why do we need this check at all? Does it make something better? Are there tests that fail if it is removed? It would be interesting to see what tests fail and how do they fail.